From 2fca80eec2521e951b389335e08edd495f9e1bd7 Mon Sep 17 00:00:00 2001
From: kirankumarb07 <kirankumarb@ami.com>
Date: Tue, 21 Mar 2023 19:19:02 +0530
Subject: [PATCH] Fix User Management Page issues title and Policy

This patchset will fix the following issues in the user management page

1. "lockoutThreshold" and "lockoutDuration" values are not shown in the
account policy model when we navigate to any other page and come back
to the user management page account policy model.
Fix - The "lockoutThreshold" and "lockoutDuration" values are monitored
and updated in the model window by using the watch. When navigated from
another page, the watch is not updating the values, so the default
values are displayed. Initialized the value with an empty object, so
the watch will get triggered and update the fields with proper values.

2. The User Management Page title is not translated properly. The
router names are used to map each page's language objects. For the user
management page, the router name and the object value in the locale
file are different. Due to that, the title did not get translated
properly.
Fix - Changed the user management page title to match the locale file's
language object key value.

Change-Id: I1582c9d528633465c65f5d78e31d3dc740252cb0
Signed-off-by: Kirankumar Ballapalli <kirankumarb@ami.com>
---
 src/router/routes.js                                          | 2 +-
 src/views/SecurityAndAccess/UserManagement/UserManagement.vue | 4 +++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/router/routes.js b/src/router/routes.js
index 3cbdabc..3ccea54 100644
--- a/src/router/routes.js
+++ b/src/router/routes.js
@@ -153,7 +153,7 @@ const routes = [
       },
       {
         path: '/security-and-access/user-management',
-        name: 'local-users',
+        name: 'user-management',
         component: UserManagement,
         meta: {
           title: i18n.t('appPageTitle.userManagement'),
diff --git a/src/views/SecurityAndAccess/UserManagement/UserManagement.vue b/src/views/SecurityAndAccess/UserManagement/UserManagement.vue
index b17be86..0af271d 100644
--- a/src/views/SecurityAndAccess/UserManagement/UserManagement.vue
+++ b/src/views/SecurityAndAccess/UserManagement/UserManagement.vue
@@ -102,7 +102,7 @@
       </b-col>
     </b-row>
     <!-- Modals -->
-    <modal-settings :settings="settings" @ok="saveAccountSettings" />
+    <modal-settings :settings="setting" @ok="saveAccountSettings" />
     <modal-user
       :user="activeUser"
       :password-requirements="passwordRequirements"
@@ -158,6 +158,7 @@ export default {
     return {
       isBusy: true,
       activeUser: null,
+      setting: {},
       fields: [
         {
           key: 'checkbox',
@@ -282,6 +283,7 @@ export default {
         });
     },
     initModalSettings() {
+      this.setting = this.settings;
       this.$bvModal.show('modal-settings');
     },
     saveUser({ isNewUser, userData }) {
-- 
2.34.1

